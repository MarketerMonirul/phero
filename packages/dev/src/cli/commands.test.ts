import {
  parseClientCommand,
  parseSamenCommand,
  parseServerCommand,
} from "./commands"

describe("parseServerCommand", () => {
  function run(cmd: string) {
    return parseServerCommand(cmd.split(" "))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'samen-server --version'", () => {
      expect(run("samen-server --version")).toMatchSnapshot()
    })

    test("cmd:   'samen-server -v'", () => {
      expect(run("samen-server -v")).toMatchSnapshot()
    })

    test("cmd:   'samen-server --help'", () => {
      expect(run("samen-server --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-server -h'", () => {
      expect(run("samen-server -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-server nothing'", () => {
      expect(err("samen-server nothing")).toThrow()
    })
  })

  describe("serve", () => {
    test("cmd:   'samen-server serve --help'", () => {
      expect(run("samen-server serve --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve -h'", () => {
      expect(run("samen-server serve -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve'", () => {
      expect(run("samen-server serve")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve --verbose'", () => {
      expect(run("samen-server serve --verbose")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve --port 1234'", () => {
      expect(run("samen-server serve --port 1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve -p 1234'", () => {
      expect(run("samen-server serve -p 1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-server serve -p 1234 --verbose'", () => {
      expect(run("samen-server serve -p 1234 --verbose")).toMatchSnapshot()
    })
  })

  describe("build", () => {
    test("cmd:   'samen-server build --help'", () => {
      expect(run("samen-server build --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-server build -h'", () => {
      expect(run("samen-server build -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-server build'", () => {
      expect(run("samen-server build")).toMatchSnapshot()
    })

    test("cmd:   'samen-server build --verbose'", () => {
      expect(run("samen-server build --verbose")).toMatchSnapshot()
    })
  })
})

describe("parseClientCommand", () => {
  function run(cmd: string) {
    return parseClientCommand(cmd.split(" "))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'samen-client --version'", () => {
      expect(run("samen-client --version")).toMatchSnapshot()
    })

    test("cmd:   'samen-client -v'", () => {
      expect(run("samen-client -v")).toMatchSnapshot()
    })

    test("cmd:   'samen-client --help'", () => {
      expect(run("samen-client --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-client -h'", () => {
      expect(run("samen-client -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-client nothing'", () => {
      expect(err("samen-client nothing")).toThrow()
    })
  })

  describe("watch", () => {
    test("cmd:   'samen-client watch --help'", () => {
      expect(run("samen-client watch --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch -h'", () => {
      expect(run("samen-client watch -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch'", () => {
      expect(run("samen-client watch")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch http://localhost:1234'", () => {
      expect(run("samen-client watch http://localhost:1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch --verbose'", () => {
      expect(run("samen-client watch --verbose")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch --port 1234'", () => {
      expect(run("samen-client watch --port 1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch -p 1234'", () => {
      expect(run("samen-client watch -p 1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch http://localhost:1234 --verbose --port 4321'", () => {
      expect(
        run("samen-client watch http://localhost:1234 --verbose --port 4321"),
      ).toMatchSnapshot()
    })

    test("cmd:   'samen-client watch ../api'", () => {
      expect(err("samen-client watch ../api")).toThrow()
    })
  })

  describe("build", () => {
    test("cmd:   'samen-client build --help'", () => {
      expect(run("samen-client build --help")).toMatchSnapshot()
    })

    test("cmd:   'samen-client build -h'", () => {
      expect(run("samen-client build -h")).toMatchSnapshot()
    })

    test("cmd:   'samen-client build'", () => {
      expect(run("samen-client build")).toMatchSnapshot()
    })

    test("cmd:   'samen-client build http://localhost:1234'", () => {
      expect(run("samen-client build http://localhost:1234")).toMatchSnapshot()
    })

    test("cmd:   'samen-client build --verbose'", () => {
      expect(run("samen-client build --verbose")).toMatchSnapshot()
    })

    test("cmd:   'samen-client build http://localhost:1234 --verbose'", () => {
      expect(
        run("samen-client build http://localhost:1234 --verbose"),
      ).toMatchSnapshot()
    })
  })
})

describe("parseSamenCommand", () => {
  function run(cmd: string) {
    return parseSamenCommand(cmd.split(" "))
  }

  function err(cmd: string) {
    return () => run(cmd)
  }

  describe("common", () => {
    test("cmd:   'samen --version'", () => {
      expect(run("samen --version")).toMatchSnapshot()
    })

    test("cmd:   'samen -v'", () => {
      expect(run("samen -v")).toMatchSnapshot()
    })

    test("cmd:   'samen --help'", () => {
      expect(run("samen --help")).toMatchSnapshot()
    })

    test("cmd:   'samen -h'", () => {
      expect(run("samen -h")).toMatchSnapshot()
    })

    test("cmd:   'samen nothing'", () => {
      expect(err("samen nothing")).toThrow()
    })
  })

  describe("dev-env", () => {
    test("cmd:   'samen'", () => {
      expect(run("samen")).toMatchSnapshot()
    })
  })

  describe("client", () => {
    test("cmd:   'samen client watch'", () => {
      expect(run("samen client watch")).toMatchSnapshot()
    })

    test("cmd:   'samen client watch --port 1234'", () => {
      expect(run("samen client watch --port 1234")).toMatchSnapshot()
    })
  })

  describe("server", () => {
    test("cmd:   'samen server serve'", () => {
      expect(run("samen server serve")).toMatchSnapshot()
    })

    test("cmd:   'samen server serve --port 1234'", () => {
      expect(run("samen server serve --port 1234")).toMatchSnapshot()
    })
  })
})
