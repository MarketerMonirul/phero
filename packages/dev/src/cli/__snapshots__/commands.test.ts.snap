// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseClientCommand build cmd:   'samen-client build --help' 1`] = `
Object {
  "command": "build",
  "name": "help",
}
`;

exports[`parseClientCommand build cmd:   'samen-client build --verbose' 1`] = `
Object {
  "name": "build",
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand build cmd:   'samen-client build -h' 1`] = `
Object {
  "command": "build",
  "name": "help",
}
`;

exports[`parseClientCommand build cmd:   'samen-client build http://localhost:1234 --verbose' 1`] = `
Object {
  "name": "build",
  "server": Object {
    "url": "http://localhost:1234",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand build cmd:   'samen-client build http://localhost:1234' 1`] = `
Object {
  "name": "build",
  "server": Object {
    "url": "http://localhost:1234",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand build cmd:   'samen-client build' 1`] = `
Object {
  "name": "build",
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand common cmd:   'samen-client --help' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseClientCommand common cmd:   'samen-client --version' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseClientCommand common cmd:   'samen-client -h' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseClientCommand common cmd:   'samen-client -v' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch --help' 1`] = `
Object {
  "command": "watch",
  "name": "help",
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch --port 1234' 1`] = `
Object {
  "name": "watch",
  "port": 1234,
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch --verbose' 1`] = `
Object {
  "name": "watch",
  "port": 4040,
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch -h' 1`] = `
Object {
  "command": "watch",
  "name": "help",
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch -p 1234' 1`] = `
Object {
  "name": "watch",
  "port": 1234,
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch http://localhost:1234 --verbose --port 4321' 1`] = `
Object {
  "name": "watch",
  "port": 4321,
  "server": Object {
    "url": "http://localhost:1234",
  },
  "verbose": true,
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch http://localhost:1234' 1`] = `
Object {
  "name": "watch",
  "port": 4040,
  "server": Object {
    "url": "http://localhost:1234",
  },
  "verbose": false,
}
`;

exports[`parseClientCommand watch cmd:   'samen-client watch' 1`] = `
Object {
  "name": "watch",
  "port": 4040,
  "server": Object {
    "url": "http://localhost:3030",
  },
  "verbose": false,
}
`;

exports[`parseSamenCommand client cmd:   'samen client watch --port 1234' 1`] = `
Object {
  "command": Object {
    "name": "watch",
    "port": 1234,
    "server": Object {
      "url": "http://localhost:3030",
    },
    "verbose": false,
  },
  "name": "client",
  "verbose": false,
}
`;

exports[`parseSamenCommand client cmd:   'samen client watch' 1`] = `
Object {
  "command": Object {
    "name": "watch",
    "port": 4040,
    "server": Object {
      "url": "http://localhost:3030",
    },
    "verbose": false,
  },
  "name": "client",
  "verbose": false,
}
`;

exports[`parseSamenCommand common cmd:   'samen --help' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseSamenCommand common cmd:   'samen --version' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseSamenCommand common cmd:   'samen -h' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseSamenCommand common cmd:   'samen -v' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseSamenCommand dev-env cmd:   'samen' 1`] = `
Object {
  "name": "dev-env",
  "verbose": false,
}
`;

exports[`parseSamenCommand server cmd:   'samen server serve --port 1234' 1`] = `
Object {
  "command": Object {
    "name": "serve",
    "port": 1234,
    "verbose": false,
  },
  "name": "server",
  "verbose": false,
}
`;

exports[`parseSamenCommand server cmd:   'samen server serve' 1`] = `
Object {
  "command": Object {
    "name": "serve",
    "port": 3030,
    "verbose": false,
  },
  "name": "server",
  "verbose": false,
}
`;

exports[`parseServerCommand build cmd:   'samen-server build --help' 1`] = `
Object {
  "command": "build",
  "name": "help",
}
`;

exports[`parseServerCommand build cmd:   'samen-server build --verbose' 1`] = `
Object {
  "name": "build",
  "verbose": true,
}
`;

exports[`parseServerCommand build cmd:   'samen-server build -h' 1`] = `
Object {
  "command": "build",
  "name": "help",
}
`;

exports[`parseServerCommand build cmd:   'samen-server build' 1`] = `
Object {
  "name": "build",
  "verbose": false,
}
`;

exports[`parseServerCommand common cmd:   'samen-server --help' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseServerCommand common cmd:   'samen-server --version' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseServerCommand common cmd:   'samen-server -h' 1`] = `
Object {
  "command": undefined,
  "name": "help",
}
`;

exports[`parseServerCommand common cmd:   'samen-server -v' 1`] = `
Object {
  "name": "version",
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve --help' 1`] = `
Object {
  "command": "serve",
  "name": "help",
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve --port 1234' 1`] = `
Object {
  "name": "serve",
  "port": 1234,
  "verbose": false,
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve --verbose' 1`] = `
Object {
  "name": "serve",
  "port": 3030,
  "verbose": true,
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve -h' 1`] = `
Object {
  "command": "serve",
  "name": "help",
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve -p 1234 --verbose' 1`] = `
Object {
  "name": "serve",
  "port": 1234,
  "verbose": true,
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve -p 1234' 1`] = `
Object {
  "name": "serve",
  "port": 1234,
  "verbose": false,
}
`;

exports[`parseServerCommand serve cmd:   'samen-server serve' 1`] = `
Object {
  "name": "serve",
  "port": 3030,
  "verbose": false,
}
`;
